### PR TITLE
add get_params_to_unitarytensor_func

### PR DESCRIPTION
**Pull request by SamDuffield**

### CODE REVIEW

File: tests/test_statetensor.py [hunk 1]
```diff
@@ -18,13 +18,21 @@ def test_H():
     assert jnp.all(jnp.abs(st.flatten() - true_sv) < 1e-5)
     assert jnp.all(jnp.abs(st_jit.flatten() - true_sv) < 1e-5)
 
+    param_to_unitary = qujax.get_params_to_unitarytensor_func(gates, qubits, param_inds)
+    unitary = param_to_unitary().reshape(2, 2)
+    unitary_jit = jit(param_to_unitary)().reshape(2, 2)
+    zero_sv = jnp.zeros(2).at[0].set(1)
+    assert jnp.all(jnp.abs(unitary @ zero_sv - true_sv) < 1e-5)
+    assert jnp.all(jnp.abs(unitary_jit @ zero_sv - true_sv) < 1e-5)
```
Comment (gamatos) [inline tests/test_statetensor.py:26]:
```suggestion
    assert jnp.allclose(unitary @ zero_sv, true_sv)
    assert jnp.allclose(unitary_jit @ zero_sv, true_sv)
```

File: tests/test_statetensor.py [hunk 2]
```diff
@@ -18,13 +18,21 @@ def test_H():
     assert jnp.all(jnp.abs(st.flatten() - true_sv) < 1e-5)
     assert jnp.all(jnp.abs(st_jit.flatten() - true_sv) < 1e-5)
```
Comment (gamatos) [inline tests/test_statetensor.py:19]:
```suggestion
    assert jnp.allclose(unitary @ zero_sv, true_sv)
    assert jnp.allclose(unitary_jit @ zero_sv, true_sv)
```

File: tests/test_statetensor.py [hunk 3]
```diff
@@ -33,14 +41,22 @@ def test_H_redundant_qubits():
     assert st.size == true_sv.size
     assert jnp.all(jnp.abs(st.flatten() - true_sv) < 1e-5)
 
+    param_to_unitary = qujax.get_params_to_unitarytensor_func(gates, qubits, param_inds, n_qubits)
+    unitary = param_to_unitary().reshape(2 ** n_qubits, 2 ** n_qubits)
+    unitary_jit = jit(param_to_unitary)().reshape(2 ** n_qubits, 2 ** n_qubits)
+    zero_sv = jnp.zeros(2 ** n_qubits).at[0].set(1)
+    assert jnp.all(jnp.abs(unitary @ zero_sv - true_sv) < 1e-5)
+    assert jnp.all(jnp.abs(unitary_jit @ zero_sv - true_sv) < 1e-5)
```
Comment (gamatos) [inline tests/test_statetensor.py:49]:
```suggestion
    assert jnp.allclose(unitary @ zero_sv, true_sv)
    assert jnp.allclose(unitary_jit @ zero_sv, true_sv)
```

File: tests/test_statetensor.py [hunk 4]
```diff
@@ -33,14 +41,22 @@ def test_H_redundant_qubits():
     assert st.size == true_sv.size
     assert jnp.all(jnp.abs(st.flatten() - true_sv) < 1e-5)
```
Comment (gamatos) [inline tests/test_statetensor.py:42]:
```suggestion
    assert jnp.allclose(st.flatten(), true_sv)
```

File: tests/test_statetensor.py [hunk 5]
```diff
@@ -50,6 +66,14 @@ def test_CX_Rz_CY():
...
```
Comment (gamatos) [inline tests/test_statetensor.py:67]:
```suggestion
    assert jnp.allclose(st.flatten(), true_sv)
```

File: tests/test_statetensor.py [hunk 6]
```diff
@@ -50,6 +66,14 @@ def test_CX_Rz_CY():
...
 
+    n_qubits = 3
+    param_to_unitary = qujax.get_params_to_unitarytensor_func(gates, qubits, param_inds, n_qubits)
+    unitary = param_to_unitary(param).reshape(2 ** n_qubits, 2 ** n_qubits)
+    unitary_jit = jit(param_to_unitary)(param).reshape(2 ** n_qubits, 2 ** n_qubits)
+    zero_sv = jnp.zeros(2 ** n_qubits).at[0].set(1)
+    assert jnp.all(jnp.abs(unitary @ zero_sv - true_sv) < 1e-5)
+    assert jnp.all(jnp.abs(unitary_jit @ zero_sv - true_sv) < 1e-5)
```
Comment (gamatos) [inline tests/test_statetensor.py:75]:
```suggestion
    assert jnp.allclose(unitary @ zero_sv, true_sv) 
    assert jnp.allclose(unitary_jit @ zero_sv, true_sv)
```

File: tests/test_statetensor.py [hunk 7]
```diff
@@ -68,4 +92,3 @@ def test_stacked_circuits():
     assert jnp.all(jnp.abs(st2.flatten() - all_zeros_sv) < 1e-5)
     assert jnp.all(jnp.abs(st2_2.flatten() - all_zeros_sv) < 1e-5)
```
Comment (gamatos) [inline tests/test_statetensor.py:93]:
```suggestion
    assert jnp.allclose(st2.flatten(), all_zeros_sv)
    assert jnp.allclose(st2_2.flatten(), all_zeros_sv)
```

**Review by gamatos — CHANGES_REQUESTED** [review]
Looks good, I would change the assertions in the tests from `jnp.all` to `jnp.allclose` though, it is more succint and has good default values for the relative and absolute tolerance (the current `1e-5` value for an absolute tolerance might be a bit high).

**Comment by SamDuffield** [comment]
> Looks good, I would change the assertions in the tests from `jnp.all` to `jnp.allclose` though, it is more succint and has good default values for the relative and absolute tolerance (the current `1e-5` value for an absolute tolerance might be a bit high).

Agreed! Much more readable

**Review by gamatos — APPROVED** [review]

**Pull request merged**